### PR TITLE
Adding Virus_total_key argument

### PR DESCRIPTION
**Pull request by PiranhaSa**
In this PR I've added the support for virustotal api key, so subfinder can use it . we can do that by adding the key to the provider-config.yaml conf file. 

From inside the agent:
Screenshot from 2024-12-06 11-59-22

As you can see here, the virustotal API is called: 
image
image

### CODE REVIEW

File: agent/subfinder_agent.py [hunk 1]
```diff
@@ -24,6 +25,33 @@
 STORAGE_NAME = "agent_subfinder_storage"
 
 
+def _update_provider_config(
+    virustotal_key: str,
+    config_path: str = "/root/.config/subfinder/provider-config.yaml",
```
Comment (amine3) [inline agent/subfinder_agent.py:30]: Use a const

Comment (amine3) [reply agent/subfinder_agent.py:30]: Based on the documentation, the default is  $CONFIG/subfinder/config.yaml no?

**Review by amine3 — DISMISSED** [review]

File: agent/subfinder_agent.py (hunk 1 above)
Comment (PiranhaSa) [reply agent/subfinder_agent.py:30]: From the dockerfile I saw that we have 
image
and from their doc it doesn't say config.yaml : 
image

File: agent/subfinder_agent.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import logging
 
+from ruamel.yaml import YAML
```
Comment (Zakaria-Latif) [inline agent/subfinder_agent.py:5]:
```suggestion
from ruamel import yaml 
```

Comment (PiranhaSa) [reply agent/subfinder_agent.py]: The name of the package is ruamel.yaml so I did change it to be import ruamel.yaml

**Review by nmasdoufi-ol — COMMENTED** [review]
LGTM

File: tests/subfinder_agent_test.py [hunk 3]
```diff
@@ -72,3 +80,46 @@ def testAgentSubfinder_whenMaxSubDomainsSet_emitsBackFindings(
...
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test that the provider configuration handles a write error correctly and logs the failure."""
+    mocker.patch("ruamel.yaml.main.YAML.dump", side_effect=FileNotFoundError)
+
+    sub_agent.update_provider_config(
+        "existing_key", str(pathlib.Path(__file__).parent / "provider-config.yaml")
+    )
+
+    assert "Failed to write configuration file" in caplog.text
```
Comment (3asm) [inline tests/subfinder_agent_test.py:125]: I don't see unit test for the `update_provider_config` when the write is working?

File: agent/subfinder_agent.py [hunk 4]
```diff
@@ -33,6 +65,12 @@ def __init__(
         agent_settings: runtime_definitions.AgentSettings,
     ) -> None:
         agent.Agent.__init__(self, agent_definition, agent_settings)
+
+        virustotal_key = self.args.get("virustotal_key")
+        if virustotal_key is not None:
+            logger.info("Updating configuration with VirusTotal API key.")
+            update_provider_config(virustotal_key)
```
Comment (3asm) [inline agent/subfinder_agent.py:72]: The name doesn't match what the method is doing. The method is just about virustotal.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**